internal/server/router/credit_balance.go

```diff
@@ -0,0 +1,51 @@
+package router
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/go-chi/render"
+
+	"github.com/openmeterio/openmeter/api"
+	"github.com/openmeterio/openmeter/pkg/contextx"
+	"github.com/openmeterio/openmeter/pkg/models"
+)
+
+// Get credit balance, GET /api/v1/credit-balance/{subject}
+func (a *Router) GetCreditBalance(w http.ResponseWriter, r *http.Request, subject string, params api.GetCreditBalanceParams) {
+	ctx := contextx.WithAttr(r.Context(), "operation", "GetCreditBalance")
+	namespace := a.config.NamespaceManager.GetDefaultNamespace()
+	cutline := time.Now()
+
+	if params.Time != nil {
+		cutline = *params.Time
+	}
+
+	// Get high watermark for credit
+	hw, err := a.config.CreditConnector.GetHighWatermark(ctx, namespace, subject)
+	if err != nil {
+		a.config.ErrorHandler.HandleContext(ctx, err)
+		models.NewStatusProblem(ctx, err, http.StatusInternalServerError).Respond(w, r)
+		return
+	}
+
+	// Check if query time is after high watermark
+	if cutline.Before(hw.Time) {
```

Comment:
This can also be problematic if the reset call was received by by a machine whose clock is ahead of this machine and a GetCreditBalance is issued right after.